sotodlib/site_pipeline/make_filterbin_map.py

```diff
@@ -0,0 +1,1113 @@
+    # and this should be good enough.
+    nphi     = utils.nint(np.abs(360/ref_wcs.wcs.cdelt[0]))
+    widths   = pixboxes[:,1,0]-pixboxes[:,0,0]
+    pixboxes[:,0,0] = utils.rewind(pixboxes[:,0,0],
+                                   ref=pixboxes[0,0,0],
+                                   period=nphi)
+    pixboxes[:,1,0] = pixboxes[:,0,0] + widths
+    # It's now safe to find the total pixel bounding box
+    union_pixbox = np.array([np.min(pixboxes[:,0],0)-pad,np.max(pixboxes[:,1],0)
+                             +pad])
+    # Use this to construct the output geometry
+    shape = union_pixbox[1]-union_pixbox[0]
+    wcs   = ref_wcs.deepcopy()
+    wcs.wcs.crpix -= union_pixbox[0,::-1]
+    if return_pixboxes: return shape, wcs, pixboxes
+    else: return shape, wcs
+
+
+def ptp_cuts(aman, signal_name='dsT', kurtosis_threshold=5):
+    while True:
+        if aman.dets.count > 0:
+            ptps = np.ptp(aman[signal_name], axis=1)
+        else:
+            break
+        kurtosis_ptp = kurtosis(ptps)
+        if kurtosis_ptp < kurtosis_threshold:
+            break
+        else:
+            max_is_bad_factor = np.max(ptps)/np.median(ptps)
+            min_is_bad_factor = np.median(ptps)/np.min(ptps)
+            if max_is_bad_factor > min_is_bad_factor:
+                aman.restrict('dets', aman.dets.vals[ptps < np.max(ptps)])
+            else:
+                aman.restrict('dets', aman.dets.vals[ptps > np.min(ptps)])
+
+
+def wrap_info(obs, site):
+    obs.wrap("weather", np.full(1, "toco"))
+    obs.wrap("site",    np.full(1, site))
+
+    # Union of flags into glitch_flags.
+    # Use this if not using pre-process database:
+    # (glitch_flags full of 0s when we don't have the preprocess data base)
+    obs.flags.wrap('glitch_flags', so3g.proj.RangesMatrix.zeros(obs.shape[:2]),
+                   [(0, 'dets'), (1, 'samps')])
+    # else:
+    #obs.flags.wrap('glitch_flags', obs.preprocess.turnaround_flags.turnarounds
+    #               + obs.preprocess.jumps_2pi.jump_flag + obs.preprocess.glitches.glitch_flags, )
+    ## TODO add --preprocess to args
+    return
+
+
+def get_detector_cuts_flags(obs, rfrac_range=(0.05,0.9), psat_range=(0,20)):
+    try:
+        flags.get_turnaround_flags(obs) 
+    except Exception as e:
+        return False
+    flags.get_det_bias_flags(obs, rfrac_range=rfrac_range, psat_range=psat_range)
+    bad_dets = has_all_cut(obs.flags.det_bias_flags)
+    obs.restrict('dets', obs.dets.vals[~bad_dets])
+    #if obs.dets.count<=1: return obs # check if I cut all the detectors after the det bias flags
+    return True
+            
+def select_data(obs,
+                det_left_right, det_in_out, det_upper_lower):
+
+    # Restrict non optical detectors, which have nans in their focal plane
+    # coordinates and will crash the mapmaking operation.
+    obs.restrict('dets', obs.dets.vals[obs.det_info.wafer.type == 'OPTC'])
+
+    splits.det_splits_relative(obs, det_left_right=det_left_right,
```

Comment:
I think we probably want to do a relative detector split right before we go into mapmaking to make sure our splits are 50/50. This will probably change in the future but in the mean time lets move this line to the end of the `calibrate_obs_otf` function.

Reply:
Done